src/proxy/server.rs

```diff
@@ -70,7 +70,7 @@ pub struct Source {
     pub remote: SocketAddr,
     pub local: SocketAddr,
     pub orig_dst: Option<SocketAddr>,
-    pub tls_status: tls::Status,
+    pub tls_peer: tls::ConditionalIdentity,
```

Comment:
If I understand correctly, this means that we can't describe the situation where TLS is established and a client ID is not provided. This case is important, especially for the service that has to bootstrap certificates -- proxies will have to be able to connect to this service before they have an identity.

I think we'll need to make the ConditionalIdentity type that is more like:

```rust
type ClientIdentity = Conditional<Identity, ReasonForNoClientIdentity>;
type ServerStatus = Conditional<ClientIdentity, ReasonForNoTls>;
```

Reply:
After taking a stab at this, it's a little complex; and, since much of the internals of this are going to have to change in an upcoming release, it's probably best to merge this as-is and address these concerns later.

Reply:
I may be missing some context, but as I was reading through the TLS code, it appeared to me that TLS couldn't be established without an identity... 🤷‍♂️